### ISSUE TITLE
Support Client ID as an alternative input to Client Secret

### ISSUE THREAD

**Issue by cloudshiftchris** [actions/create-github-app-token#136]
GitHub now [recommends](https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app) using the clientId instead of the appId/privateKey

Augment the options to allow clientId/clientSecret, validating that either this pair or appId/privateKey are provided, and pass these options along in the call to `createAppAuth`.

**Comment by gr2m** [comment]
Just to clarify, you cannot use the Client Secret, the private key is still required. But the OAuth Client ID is now recommended to be used as a substitute to the numeric app ID for github.com, and probably GitHub Enterprise Server 3.14+

**Comment by joebowbeer** [comment]
@cloudshiftchris edit title to clarify that support is sought for using clientId instead of appId?

@gr2m are you saying this is already supported via https://github.com/octokit/auth-app.js/pull/606 ?

**Comment by gr2m** [comment]
> @gr2m are you saying this is already supported via [octokit/auth-app.js#606](https://github.com/octokit/auth-app.js/pull/606) ?

yes

**Comment by BeyondEvil** [comment]
So does this mean that we can pass in client ID instead of app ID, and this issue can be closed... or? @gr2m

**Comment by joebowbeer** [comment]
I think this should be closed. I have used clientId in place of appId

**Comment by gr2m** [comment]
we should document the fact that client ID can be used as an alternative to the app ID, then we can close this issue

**Comment by jeffwidman** [comment]
A somewhat self-documenting way to do this would be to add a new `client-id` parameter:

```
with:
  client-id: <clientID>
```

Then `app-id` could be deprecated and removed in a future version.

**Comment by jeffwidman** [comment]
Makes sense--do you want to create a tracking issue for "v3 breaking changes list" so that you don't lose track of this one?
